rover_auxiliary/src/ipcamera_api.py

```diff
@@ -0,0 +1,322 @@
+            self.source_token = self.profile.VideoSourceConfiguration.SourceToken
+
+        except TimeoutException:
+            self.logger.error("Camera initialization timed out")
+        except Exception as e:
+            self.logger.error(f"Failed to initialize the camera: {str(e).split(':')[-1].strip()}")
+        finally:
+            signal.alarm(0)  
+
+    def set_onvif_param(self, param: str, value: str, extra: tuple=None):
+        try:       
+            encoder_config = self.profile.VideoEncoderConfiguration
+            imaging_settings = self.imaging_service.GetImagingSettings(self.source_token)
+
+            if extra:
+                encoder_config.Resolution.Width = extra[0]
+                encoder_config.Resolution.Height = extra[1]
+
+            if self.ONVIF_PARAMS[param] == 'imaging':
+                setattr(imaging_settings, param, value)
+                self.imaging_service.SetImagingSettings({
+                    'VideoSourceToken': self.source_token,
+                    'ImagingSettings': imaging_settings,
+                    'ForcePersistence': True
+                })
+            elif self.ONVIF_PARAMS[param] == 'video':
+                setattr(encoder_config.RateControl, param, value)
+                self.media_service.SetVideoEncoderConfiguration({
+                    'Configuration': encoder_config,
+                    'ForcePersistence': True
+                })
+            self.logger.info(f"{param} updated to {value}")
+        except Exception as e:
+            self.logger.error(f"Failed to set ONVIF parameter {param}: {e}")
+           
+    def set_http_param(self, param: str, value: str, extra: str = None):
+        try:
+            headers = {'Authorization': 'Basic YWRtaW46YWRtaW4='}
+            form_data = {
+                'flag': self.HTTP_PARAMS[param][1] if 'IR' not in param else '',
+                self.HTTP_PARAMS[param][0]: value,
+                self.HTTP_PARAMS[param][0].replace('Text', ''): extra if extra else ''
+            }
+
+            url = f"http://{self.ip}/form/{'IRset' if 'IR' in param else 'CameraSet'}"
+            response = requests.post(url, data=form_data, headers=headers)
+
+            if response.status_code == 200:
+                self.logger.info(f"{param} updated to {value}")
+            else:
+                self.logger.error(f"Failed to update {param} via HTTP. Status: {response.status_code}")
+        except Exception as e:
+            self.logger.error(f"Failed to set parameter {param}: {e}")
+
+    # ======================================================================================================================= #
+    #                                                      API METHODS                                                        #
+    # ======================================================================================================================= #
+
```

Comment:
Can you add checks for all arguments/input values to make sure they are valid for our cams?